### DISCUSSION TITLE
WebXR

### DISCUSSION THREAD

**Discussion by InfoLibre** [overte-org/overte#987] [General]
Is overte using WebXR and WebGPU?

**Comment by JulianGro** [comment]
No it isn't.
Currently, Overte uses OpenVR and OpenGL. However, we are switching to OpenXR and Vulkan. (The switch to OpenXR is almost done.)

**Comment by InfoLibre** [comment]
But if one wants to use Overte on Web with VR, WebXR is needed. And perhaps WebGPU.

**Reply by ksuprynowicz** [reply]
Current client is kinda like a 3D web browser in itself (including heavy use of JS, HTML and even Web Assembly support), and due to several different solutions it uses would be almost impossible to make it run inside a web browser. WebGPU and WebXR are not the only obstacles here, other problems include heavy dependencies that would need to be ported, and how current scripting system works. Desktop client also uses UDP, which is not available for applications running inside a web browser.
Writing a web client would be very welcome, but due to entirely different requirements it would have to be much simpler than current client. Our server does have Web API that could be used for connectivity. It would be a good idea to review and merge newer improvements to Web API that were done in Vircadia for this.
An excellent example of how to write performant web client for social VR is Matrix Third Room (which is sadly not active anymore). Their client could run at full framerate on standalone Quest Pro.

**Comment by InfoLibre** [comment]
You could write your client above https://www.chromium.org/Home/, like STEAM, Brave, Edge, Chrome. It includes WebXR.

**Comment by JulianGro** [comment]
Closing this, as @ksuprynowicz has already explained all the main points.

Too long, didn't read: Replacing Overte with a web application is unrealistic and we won't do it. 

If anyone ever creates a web client for Overte, feel free to ask them for WebXR support.
